src/main_benchmark.cpp

```diff
@@ -0,0 +1,414 @@
+        if (!dimOpt.has_value()) {
+            std::cout << __LINE__ << " " << s << std::endl;
+            throw std::invalid_argument("Invalid shape argument");
+        }
+                                   return dimOpt.value(); });
+    return shape;
+}
+
+volatile sig_atomic_t shutdown_request = 0;
+
+static void onInterrupt(int status) {
+    shutdown_request = 1;
+}
+
+static void onTerminate(int status) {
+    shutdown_request = 1;
+}
+
+static void onIllegal(int status) {
+    shutdown_request = 2;
+}
+
+static void installSignalHandlers() {
+    static struct sigaction sigIntHandler;
+    sigIntHandler.sa_handler = onInterrupt;
+    sigemptyset(&sigIntHandler.sa_mask);
+    sigIntHandler.sa_flags = 0;
+    sigaction(SIGINT, &sigIntHandler, NULL);
+
+    static struct sigaction sigTermHandler;
+    sigTermHandler.sa_handler = onTerminate;
+    sigemptyset(&sigTermHandler.sa_mask);
+    sigTermHandler.sa_flags = 0;
+    sigaction(SIGTERM, &sigTermHandler, NULL);
+
+    static struct sigaction sigIllHandler;
+    sigIllHandler.sa_handler = onIllegal;
+    sigemptyset(&sigIllHandler.sa_mask);
+    sigIllHandler.sa_flags = 0;
+    sigaction(SIGILL, &sigIllHandler, NULL);
+}
+
+using shape_t = std::vector<size_t>;
+
+OVMS_InferenceRequest* prepareRequest(const std::string& servableName, uint32_t servableVersion, OVMS_DataType datatype, const shape_t& shape, const std::string& inputName, const void* data) {
+    OVMS_InferenceRequest* request{nullptr};
+    OVMS_InferenceRequestNew(&request, servableName.c_str(), servableVersion);
+    OVMS_InferenceRequestAddInput(request, inputName.c_str(), datatype, shape.data(), shape.size());
+    size_t elementsCount = std::accumulate(shape.begin(), shape.end(), 1, std::multiplies<size_t>());
+    OVMS_InferenceRequestInputSetData(request, inputName.c_str(), data, sizeof(float) * elementsCount, OVMS_BUFFERTYPE_CPU, 0);  // TODO sizeof
+    return request;
+}
+
+void triggerInferenceInALoop(
+    std::future<void>& startSignal,
+    std::future<void>& stopSignal,
+    const size_t niterPerThread,
+    size_t& wholeThreadTimeUs,
+    double& averageWholeLatency,
+    double& averagePureLatency,
+    OVMS_Server* server,
+    OVMS_InferenceRequest* request) {
+    OVMS_InferenceResponse* response{nullptr};
+    std::vector<uint64_t> latenciesWhole(niterPerThread);
+    std::vector<uint64_t> latenciesPure(niterPerThread);
+    startSignal.get();
+    auto workloadStart = std::chrono::high_resolution_clock::now();
+    size_t iter = niterPerThread;
+    while (iter-- > 0) {
+        // stopSignal will be used with ctrl-c app stopping or with total requestCount
+        // stopSignal.wait_for(std::chrono::milliseconds(0));
+        auto iterationWholeStart = std::chrono::high_resolution_clock::now();
+        auto iterationPureStart = std::chrono::high_resolution_clock::now();
+        // aternatively we are changing request
+        OVMS_Inference(server, request, &response);
```

Comment:
Should we at least catch error from OVMS_Inference and delete it and stop the workload?